### PR TITLE
Fix tmp sgpr allocation to avoid over-writing values (alpha)

### PR DESCRIPTION
**Pull request by AlexBrownAMD**

### CODE REVIEW

File: Tensile/Tests/pre_checkin/mfma/dgemm_alpha1_beta0_sgpr.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      ForkParameters:
+        - AssertMinApproxSize: [3]
+        - DepthU: [16]
+        - EdgeType: ["ShiftPtr"]
+        - ExpandPointerSwap: [0]
+        - GlobalReadVectorWidth: [2]
+        - GroupLoadStore: [1]
+        - InnerUnroll: [2]
+        - LocalReadVectorWidth: [1]
+        - MACInstruction: ["MAD"]
```
Comment (nakajee) [inline Tensile/Tests/pre_checkin/mfma/dgemm_alpha1_beta0_sgpr.yaml:41]: Could you remove this line?

File: Tensile/Tests/pre_checkin/mfma/dgemm_alpha1_beta0_sgpr.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  # PrintSolutionRejectionReason: True
+  # BufferOffsetB: 536877696
+  DataInitTypeAlpha: 1
+  DataInitTypeBeta: 0
+
+BenchmarkProblems:
+  ########################################
+  # NT - standard
+  ########################################
+  - # sgemm TN
```
Comment (nakajee) [inline Tensile/Tests/pre_checkin/mfma/dgemm_alpha1_beta0_sgpr.yaml:17]: Comment should be dgemm NN

File: Tensile/KernelWriterAssembly.py [hunk 3]
```diff
@@ -6844,7 +6844,7 @@ def openSumAtLeastUnroll(self, kernel, prefetch, isOptNLL, isPap):
         return ""
 
       skipOptNLL = self.getNamedLabel("OptNLL_End")
-      tmpSgpr = self.getTmpSgpr(2).idx()
+      tmpSgpr = self.getTmpSgpr(4).idx()
```
Comment (nakajee) [inline Tensile/KernelWriterAssembly.py:6847]: Please make sure rocblas build works.

File: Tensile/Tests/pre_checkin/mfma/dgemm_alpha1_beta0_sgpr.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  BoundsCheck: False
+  # KernelTime: True
+  # PrintSolutionRejectionReason: True
+  # BufferOffsetB: 536877696
+  DataInitTypeAlpha: 1
+  DataInitTypeBeta: 0
+
+BenchmarkProblems:
+  ########################################
+  # NT - standard
```
Comment (nakajee) [inline Tensile/Tests/pre_checkin/mfma/dgemm_alpha1_beta0_sgpr.yaml:15]: should be NN

File: Tensile/KernelWriterAssembly.py (hunk 3 above)
Comment (AlexBrownAMD) [reply Tensile/KernelWriterAssembly.py:6847]: rocblas build worked, and gemm tests ran overnight on gfx90a

File: Tensile/Tests/pre_checkin/mfma/dgemm_alpha1_beta0_sgpr.yaml (hunk 4 above)
Comment (AlexBrownAMD) [reply Tensile/Tests/pre_checkin/mfma/dgemm_alpha1_beta0_sgpr.yaml:15]: Updated

**Review by nakajee — APPROVED** [review]
Looks good!

**Pull request merged**